src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs

```diff
@@ -531,6 +532,10 @@ impl SharedBufferBatch {
         self.inner.range_tombstone_list.clone()
     }
 
+    pub fn get_delete_range_tombstones_for_query(&self) -> Vec<DeleteRangeTombstone> {
```

Comment:
Seems the above `get_delete_range_tombstones` is reserved for its using in `compact_shared_buffer`. Can we unify `range_tombstone_list` and `range_tombstone_list_for_query` to a single one? (Maybe in a follow-up refactor PR)

Reply:
> Seems the above `get_delete_range_tombstones` is reserved for its using in `compact_shared_buffer`. Can we unify `range_tombstone_list` and `range_tombstone_list_for_query` to a single one? (Maybe in a follow-up refactor PR)

Can be done later.
5aee4b37dff3c75269b0e7835791dcd0

Reply:
> > Seems the above `get_delete_range_tombstones` is reserved for its using in `compact_shared_buffer`. Can we unify `range_tombstone_list` and `range_tombstone_list_for_query` to a single one? (Maybe in a follow-up refactor PR)
> 
> Can be done later. 5aee4b37dff3c75269b0e7835791dcd0

typo? Should be a link to some PR or issue?

Reply:
I found that there is a PR (https://github.com/risingwavelabs/risingwave/pull/8927) to optimize the delete range implementation overall. Do you plan to unify the `range_tombstone_list` in that PR? @soundOfDestiny

Reply:
> I found that there is a PR (#8927) to optimize the delete range implementation overall. Do you plan to unify the two `range_tombstone_list` in that PR? @soundOfDestiny

I have also written a #8927 version of this PR.

Reply:
Ok, we can merge current PR as a quick fix of the bug

Reply:
> Seems the above `get_delete_range_tombstones` is reserved for its using in `compact_shared_buffer`. Can we unify `range_tombstone_list` and `range_tombstone_list_for_query` to a single one? (Maybe in a follow-up refactor PR)

fixed in #9255